src/DotNetWorker/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -56,6 +57,10 @@ public static IFunctionsWorkerApplicationBuilder AddFunctionsWorker(this IServic
             // Function Execution Contexts
             services.AddSingleton<IFunctionContextFactory, DefaultFunctionContextFactory>();
 
+            // Invocation Features
+            services.TryAddSingleton<IInvocationFeaturesFactory, DefaultInvocationFeaturesFactory>();
+            services.AddSingleton<IInvocationFeatureProvider, ConverterBindingFeatureProvider>();
```

Comment:
Would you expect we move stuff like the FunctionInvocation off of the context directly, and into a Feature?

Reply:
Perhaps the core logic. Still having the middleware, but potentially moving to this.